cli.py

```diff
@@ -0,0 +1,111 @@
+from dataclasses import dataclass
+from typing import Callable, Any
+
+import out
+import constants
+
+
+@dataclass
+class CliOptions:
+    force: bool = False
+    auth: bool = False
+    help: bool = False
+    project: str = ""
+
+
+@dataclass(frozen=True)
+class CliArgument:
+    name: str
+    aliases: list[str]
+    description: str
+    value_parser: Callable[[str], Any]
+
+
+cli_arguments = [
+    CliArgument("help", ["h"], "Show this help message", lambda x: bool(x)),
+    CliArgument("force", ["f"], "Force the operation", lambda x: bool(x)),
+    CliArgument(
+        "auth", ["a"], "Retype your L.I.S.T. email and password", lambda x: bool(x)
+    ),
+]
+# The default param that doesn't require a switch to be used
+cli_param = CliArgument(
+    "project",
+    [],
+    "optionally specify project name (folder name) to submit",
+    lambda x: str(x),
+)
+
+
+def find_argument(name: str) -> CliArgument:
+    for arg in cli_arguments:
+        if arg.name == name or name in arg.aliases:
+            return arg
+    raise ValueError(f"Unknown argument: {name}")
+
+
+def parse_cli_params(argv: list[str]) -> CliOptions:
+    options = CliOptions()
+
+    if len(argv) <= 1:
+        return options
+
+    parsed_param = False
+
+    i = 1
+    while i < len(argv):
+        arg = argv[i]
+        if arg.startswith("--"):
+            switch = arg[2:]
+        elif arg.startswith("-"):
+            switch = arg[1:]
+        elif not parsed_param:
+            parsed_param = True
+            setattr(options, cli_param.name, cli_param.value_parser(arg))
+            i += 1
+            continue
+        else:
+            raise ValueError(f"Unknown argument: {arg}")
```

Comment:
What is parsed_param for ? It feels meaningless.

Reply:
sorrko, ze to na tom preview nevidno, neviem ako spravit aby sa koment priradil k celej mnou oznacenej casti.

Reply:
It is used to parse the param only once